pkg/graveler/graveler.go

```diff
@@ -842,26 +855,27 @@ func (g *KVGraveler) GetRepository(ctx context.Context, repositoryID RepositoryI
 }
 
 func (g *KVGraveler) CreateRepository(ctx context.Context, repositoryID RepositoryID, storageNamespace StorageNamespace, branchID BranchID) (*Repository, error) {
-	repo := Repository{
-		StorageNamespace: storageNamespace,
-		CreationDate:     time.Now(),
-		DefaultBranchID:  branchID,
+	_, err := g.RefManager.GetRepository(ctx, repositoryID)
+	if err != nil && !errors.Is(err, ErrRepositoryNotFound) {
+		return nil, err
 	}
```

Comment:
What's the purpose of this test if it can happen anyhow during creation? I think we should expect `kv.ErrPredicateFailed` from the creation and act on that.

Reply:
We want to provide the user with informative error in case of different repository states.
For example, if repository is "in deletion". The user will know to either wait for deletion to complete and retry or call delete again